ab-testing/scripts/deploy/fastly-api.ts

```diff
@@ -0,0 +1,306 @@
+import * as env from './env.ts';
+import {
+	object,
+	string,
+	assert,
+	array,
+	nullable,
+	boolean,
+	number,
+	type,
+} from 'jsr:@superstruct/core';
+
+const dictionaryItemStruct = object({
+	service_id: string(),
+	item_key: string(),
+	item_value: string(),
+	dictionary_id: string(),
+	created_at: string(),
+	updated_at: string(),
+	deleted_at: nullable(string()),
+});
+
+type UpdateDictionaryItemRequest =
+	| {
+			item_key: string;
+			item_value: string;
+			op: 'create' | 'update' | 'upsert';
+	  }
+	| {
+			item_key: string;
+			op: 'delete';
+	  };
+
+/**
+ * Fetch data from Fastly API
+ * We're using fetch instead of the Fastly API client as it doesn't play nicely with Deno, nor is it typed
+ *
+ * @param url - The URL to fetch
+ * @param options - Fetch options
+ * @returns The parsed JSON response
+ * @throws Error if the response is not ok or if the response cannot be parsed
+ */
+const fetchFromFastly = async <T>(
+	url: string,
+	options: RequestInit = {},
+): Promise<T> => {
+	const response = await fetch(url, {
+		...options,
+		headers: {
+			...options.headers,
+			'Fastly-Key': env.FASTLY_API_TOKEN,
+		},
+	});
+	if (!response.ok) {
+		console.error(await response.text());
+		throw new Error(
+			`Failed to fetch from Fastly: ${response.status} ${response.statusText}`,
+		);
+	}
+	const data = await response.json();
+	if (!data) {
+		throw new Error(`Failed to parse response from Fastly`);
+	}
+	return data;
+};
+
+const getService = async (serviceId: string) => {
+	const service = await fetchFromFastly(
+		`https://api.fastly.com/service/${serviceId}`,
+	);
+
+	assert(
+		service,
+		type({
+			id: string(),
+			name: string(),
+			versions: array(
+				type({
+					active: boolean(),
+					number: number(),
+				}),
+			),
+		}),
+	);
+
+	return service;
+};
+
+const getDictionary = async ({
+	serviceId,
+	activeVersion,
+	dictionaryName,
+}: {
+	serviceId: string;
+	activeVersion: number;
+	dictionaryName: string;
+}) => {
+	const dictionary = await fetchFromFastly(
+		`https://api.fastly.com/service/${serviceId}/version/${activeVersion}/dictionary/${dictionaryName}`,
```

Comment:
Sorry I am going to be that person. _Could this be a const_ ?

`https://api.fastly.com/service/`